apps/prairielearn/src/tests/groupExam.test.js

```diff
@@ -0,0 +1,401 @@
+const { config } = require('../lib/config');
+const assert = require('chai').assert;
+const cheerio = require('cheerio');
+const fetch = require('node-fetch').default;
+const { step } = require('mocha-steps');
+
+const sqldb = require('@prairielearn/postgres');
+const sql = sqldb.loadSqlEquiv(__filename);
+
+const helperServer = require('./helperServer');
+const { idsEqual } = require('../lib/id');
+const { TEST_COURSE_PATH } = require('../lib/paths');
+
+let page, elemList;
+const locals = {};
+locals.helperClient = require('./helperClient');
+locals.siteUrl = 'http://localhost:' + config.serverPort;
+locals.baseUrl = locals.siteUrl + '/pl';
+locals.courseInstanceUrl = locals.baseUrl + '/course_instance/1';
+locals.assessmentsUrl = locals.courseInstanceUrl + '/assessments';
+
+const storedConfig = {};
+
+/**
+ * Switches `config` to new user, loads assessment page, and changes local CSRF token
+ * @param {Object} studentUser
+ * @param {string} assessmentUrl
+ * @param {String} authUin
+ * @param {Number} numCsrfTokens
+ */
+const switchUserAndLoadAssessment = async (studentUser, assessmentUrl, authUin, numCsrfTokens) => {
+  // Load config
+  config.authUid = studentUser.uid;
+  config.authName = studentUser.name;
+  config.authUin = authUin;
+  config.userId = studentUser.user_id;
+
+  // Load assessment
+  const res = await fetch(assessmentUrl);
+  assert.isOk(res.ok);
+  const page = await res.text();
+  locals.$ = cheerio.load(page);
+
+  // Check for CSRF tokens
+  elemList = locals.$('form input[name="__csrf_token"]');
+  assert.lengthOf(elemList, numCsrfTokens);
+  assert.nestedProperty(elemList[0], 'attribs.value');
+  locals.__csrf_token = elemList[0].attribs.value;
+  assert.isString(locals.__csrf_token);
+};
+
+const createGroup = async (groupName, csrfToken, assessmentUrl) => {
+  const form = {
+    __action: 'create_group',
+    __csrf_token: csrfToken,
+    groupName: groupName,
+  };
+  const res = await fetch(assessmentUrl, {
+    method: 'POST',
+    body: new URLSearchParams(form),
+  });
```

Comment:
Nit, there's generally no reason to declare the form separately.

```suggestion
  const res = await fetch(assessmentUrl, {
    method: 'POST',
    body: new URLSearchParams({
      __action: 'create_group',
      __csrf_token: csrfToken,
      groupName: groupName,
    }),
  });
```